smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/retries/StandardRetryIntegration.kt

```diff
@@ -0,0 +1,79 @@
+
+object RetryConfigSection : SectionId
+
+/**
+ * Adds retry wrappers around operation invocations, driven by client config (which specifies the [RetryStrategy]) and
+ * a middleware (which specifies the [RetryPolicy][aws.smithy.kotlin.runtime.retries.RetryPolicy]). This class is
+ * `open` and may be subclassed to customize the strategy and/or policy. Note that codegen which specifies a subclass of
+ * this should probably configure [replacesIntegrations] to remove the base implementation—otherwise, multiple copies of
+ * the config properties may be included in client config.
+ */
+class StandardRetryIntegration : KotlinIntegration {
+    override fun additionalServiceConfigProps(ctx: CodegenContext): List<ClientConfigProperty> =
+        listOf(
+            ClientConfigProperty {
+                val retryStrategyBlock = """
+                    run {
+                        val strategyOptions = StandardRetryStrategyOptions.Default
+                        val tokenBucket = StandardRetryTokenBucket(StandardRetryTokenBucketOptions.Default)
+                        val delayer = ExponentialBackoffWithJitter(ExponentialBackoffWithJitterOptions.Default)
+                        StandardRetryStrategy(strategyOptions, tokenBucket, delayer)
+                    }
+                """.trimIndent()
+
+                symbol = buildSymbol {
+                    name = "RetryStrategy"
+                    namespace(KotlinDependency.CORE, "retries")
+                    nullable = false
+                    reference(RuntimeTypes.Core.Retries.Impl.StandardRetryStrategy, ContextOption.USE)
+                    reference(RuntimeTypes.Core.Retries.Impl.StandardRetryStrategyOptions, ContextOption.USE)
+                    reference(RuntimeTypes.Core.Retries.Impl.StandardRetryTokenBucket, ContextOption.USE)
+                    reference(RuntimeTypes.Core.Retries.Impl.StandardRetryTokenBucketOptions, ContextOption.USE)
+                    reference(RuntimeTypes.Core.Retries.Impl.ExponentialBackoffWithJitter, ContextOption.USE)
+                    reference(RuntimeTypes.Core.Retries.Impl.ExponentialBackoffWithJitterOptions, ContextOption.USE)
+                }
+                name = "retryStrategy"
+                documentation = """
+                    The [RetryStrategy] implementation to use for service calls. All API calls will be wrapped by the
+                    strategy.
+                """.trimIndent()
+                constantValue = retryStrategyBlock
+            }
+        )
+
+    override fun customizeMiddleware(
+        ctx: ProtocolGenerator.GenerationContext,
+        resolved: List<ProtocolMiddleware>,
+    ): List<ProtocolMiddleware> = resolved + middleware
+
+    private val middleware = object : HttpFeatureMiddleware() {
+        override val name: String = "RetryFeature"
+
+        override fun renderConfigure(writer: KotlinWriter) {
+            writer.declareSection(RetryConfigSection) {
```

Comment:
this isn't quite what I had in mind..

There's two things here. The (1) config property and the (2) middleware. 

* For (1) just add the config property to the list of [default properties](https://github.com/awslabs/smithy-kotlin/blob/bd016aea94abafa48a977c1574398e04384144db/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/ClientConfigGenerator.kt#L34) if we are taking a stance that all SDK clients should be generated with a retry strategy (and I think we should). We have a way to override this if needed already to remove default props. No need for this to be in an integration.

* For (2) you don't need a `Section`. Since it's a middleware just wholesale [replace](https://github.com/awslabs/smithy-kotlin/blob/f715840b97230deb9c42ffbf920e8bbc5b63a822/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/protocol/ProtocolMiddleware.kt#L52) the thing in `aws-sdk-kotlin`. You can make it re-usable if you want or just duplicate it to a new `HttpFeatureMiddleware` type there. Since it's not a lot of code I'd probably just duplicate it personally.


I'm not going to die on this hill, if you think they belong as a `KotlinIntegration` I'll live with it. It just seems at odds with what we've done before.